source/MaterialXCore/MaterialNode.cpp

```diff
@@ -0,0 +1,151 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXCore/MaterialNode.h>
+
+namespace MaterialX
+{
+
+
+std::list<NodePtr> getShaderNodes(const NodePtr materialNode, const string& shaderType, const string& target)
+{
+    ElementPtr parent = materialNode ? materialNode->getParent() : nullptr;
```

Comment:
Nitpick: Do we really have to check for nullptr on the input? Asking for shaders on a nullptr material feels like an API user error. If there is a risk of this being null that check should be done _before_ the call is made. If sending in a nullptr is invalid so have to throw another option here would be to send in a const ref instead. That way any check has to happen before the call.